Mathlib/Topology/ContinuousMap/Algebra.lean

```diff
@@ -579,6 +579,18 @@ instance [SMul R M] [ContinuousConstSMul R M] [SMul R₁ M] [ContinuousConstSMul
 instance [SMul R M] [SMul Rᵐᵒᵖ M] [ContinuousConstSMul R M] [IsCentralScalar R M] :
     IsCentralScalar R C(α, M) where op_smul_eq_smul _ _ := ext fun _ => op_smul_eq_smul _ _
 
+instance [SMul R M] [ContinuousConstSMul R M] [Mul M] [ContinuousMul M] [IsScalarTower R M M] :
+    IsScalarTower R C(α, M) C(α, M) where
+  smul_assoc _ _ _ := ext fun _ => smul_mul_assoc ..
```

Comment:
I think that [ContinuousMap.instSMul'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/ContinuousMap/Algebra.html#ContinuousMap.instSMul%27) should be generalized to `[SMul R M] [ContinuousSMul R M]` and these instances should be generalized to `IsScalarTower R C(α, S) C(α, M)` etc.

Reply:
@eric-wieser Does `instSMul'` generate diamonds for something like `SMul C(α, M) C(α, C(α, M))`?

Reply:
Yes, that instance definitely generates diamonds, and should probably be scoped as a result.

Reply:
In that case I would prefer that:

1. the generalization / scoping of `instSMul'` is left for a future PR.
2. these instances are left here as is (since they apply to an instance which is/will be available globally.